crates/ruff_linter/src/rules/pylint/rules/global_variable_undefined.rs

```diff
@@ -0,0 +1,137 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast, Expr, Stmt, StmtGlobal};
+use ruff_python_semantic::{BindingKind, Scope, ScopeKind};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks that all `global` variables are indeed defined on module level
+///
+/// ## Why is this bad?
+/// If the module level declaration is missing, then either if was
+/// forgotten or the `global` can be omitted.
+///
+/// ## Example
+/// ```python
+/// def foo():
+///     global var  # [global-variable-undefined]
+///     var = 10
+///     print(var)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// var = 1
+///
+///
+/// def foo():
+///     global var
+///     var = 10
+///     print(var)
+/// ```
+#[violation]
+pub struct GlobalVariableUndefined {
+    name: String,
+}
+
+impl Violation for GlobalVariableUndefined {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let GlobalVariableUndefined { name } = self;
+        format!("Global variable `{name}` is undefined at the module")
+    }
+}
+
+/// PLW0601
+pub(crate) fn global_variable_undefined(checker: &mut Checker, stmt: &Stmt) {
+    if checker.semantic().current_scope().kind.is_module() {
+        return;
+    }
+
+    let Stmt::Global(StmtGlobal { names, range }) = stmt else {
+        return;
+    };
+
+    if checker
+        .semantic()
+        .current_scopes()
+        .any(Scope::uses_star_imports)
+    {
+        return;
+    }
+
+    let Some(module_scope) = checker
+        .semantic()
+        .current_scopes()
+        .find(|scope| scope.kind.is_module())
+    else {
+        return;
+    };
```

Comment:
Do you think we could use `global_scope` here instead?

```suggestion
    let module_scope = checker
        .semantic()
        .global_scope();
```